modules/spring-data-2.2-ext/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java

```diff
@@ -89,29 +96,37 @@ public IgniteRepositoryFactory(ApplicationContext ctx) {
     }
 
     /** */
-    private Ignite igniteForRepoConfig(RepositoryConfig config) {
+    private IgniteProxy igniteForRepoConfig(RepositoryConfig config) {
         try {
-            String igniteInstanceName = evaluateExpression(config.igniteInstance());
-            return (Ignite)ctx.getBean(igniteInstanceName);
+            Object igniteInstanceBean = ctx.getBean(evaluateExpression(config.igniteInstance()));
+
+            if (igniteInstanceBean instanceof Ignite)
+                return new IgniteProxyImpl((Ignite)igniteInstanceBean);
+            else if (igniteInstanceBean instanceof IgniteClient)
+                return new IgniteClientProxy((IgniteClient)igniteInstanceBean);
+
+            throw new IllegalStateException("Invalid repository configuration. The Spring Bean corresponding to the" +
+                " \"igniteInstance\" property of repository configuration must be one of the following types:" +
+                " \"org.apache.ignite.Ignite\", \"org.apache.ignite.client.IgniteClient\"");
         }
         catch (BeansException ex) {
             try {
                 String igniteConfigName = evaluateExpression(config.igniteCfg());
                 IgniteConfiguration cfg = (IgniteConfiguration)ctx.getBean(igniteConfigName);
                 try {
                     // first try to attach to existing ignite instance
-                    return Ignition.ignite(cfg.getIgniteInstanceName());
+                    return new IgniteProxyImpl(Ignition.ignite(cfg.getIgniteInstanceName()));
                 }
                 catch (Exception ignored) {
                     // nop
                 }
-                return Ignition.start(cfg);
+                return new IgniteProxyImpl(Ignition.start(cfg));
```

Comment:
We can try. But, from my point of view, it's better to use spring-boot-thin-client-autoconfigure for this purpose instead of placing the responsibility of starting client instances on the Spring Data repository.

Reply:
We have the ability to start Ignite nodes now, starting thin client is much more lightweight and I think sometimes will be helpful. Those who use spring-data not always use spring-boot.

Reply:
Done.